tests/test_repro.py

```diff
@@ -1404,3 +1405,37 @@ def test(self):
 
         expected_metrics_display = "{}: {}".format(metrics_file, metrics_value)
         self.assertIn(expected_metrics_display, self._caplog.text)
+
+
+class TestPersistentOutput(TestDvc):
```

Comment:
This is not testing running `dvc run` twice, which is where `persist` should make second run ignore build cache and actually run the command.